staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -70,6 +70,10 @@ type Options struct {
 	// See: https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
 	IssuerURL string
 
+	// IssuerHTTPProxy, if specified, is used to proxy connection when the provider
+	// can not be reached directly (for example, from a private network without access to the Internet).
+	IssuerHTTPProxy string
```

Comment:
to my knowledge, we don't have specific connection proxy settings like this elsewhere... we either use the HTTP_PROXY/HTTPS_PROXY/NO_PROXY envvars or we use the options specified in `--egress-selector-config-file` to determine how to make connections.

`--egress-selector-config-file` currently supports configurations for these contexts:
* `controlplane` (for making connections to authentication/authorization/audit/tracing endpoints)
* `etcd` (for making connections to the etcd server)
* `cluster` (for making connections to pods/services)

It also currently supports the following network configurations for those contexts:
* `Direct` (normal http connection behavior)
* `GRPC` (connect via konnectivity network proxy using grpc protocol)
* `HTTPConnect` (connect via konnectivity network proxy using http connect protocol)

Rather than adding separate proxy configuration like this PR does, I would expect to use or extend the `--egress-selector-config-file` to be able to handle your scenario. I'm not sure if the `HTTPConnect` option only works with the konnectivity network proxy, or if it would work pointed at a generic http proxy (cc @jkh52). If it doesn't work with generic network proxies, it seems reasonable to me to have an egress-selector option to point at a generic http proxy.

Reply:
Thank you for the feedback, @liggitt 
I close this PR and will look at `--egress-selector-config-file`.